src/test/java/cat/udl/eps/softarch/demo/steps/DonationStepDefs.java

```diff
@@ -0,0 +1,38 @@
+package cat.udl.eps.softarch.demo.steps;
+
+import cat.udl.eps.softarch.demo.domain.Donation;
+import cat.udl.eps.softarch.demo.domain.Donor;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+public class DonationStepDefs {
+    public static Donation donation;
+    public static Donor donor;
+    @Given("^There is a donor registered with the username \"([^\"]*)\"$")
+    public void thereIsADonor(String username){
+        donor = new Donor();
+        donor.setUsername(username);
+    }
+
+    @When("^I create a new donation$")
+    public void createDonation(){
```

Comment:
Remember, the @When should go through the API, like in the example: https://github.com/UdL-EPS-SoftArch/mycircular-api/blob/main/src/test/java/cat/udl/eps/softarch/demo/steps/CreateTransDefs.java#L79